src/server/customError.ts

```diff
@@ -0,0 +1,40 @@
+export enum CustomErrors {
+  UserAlreadyExists = 'UserAlreadyExists',
+  UserNotFound = 'UserNotFound',
+  InternalServerError = 'InternalServerError',
+  InvalidLoginCredentials = 'InvalidLoginCredentials',
+}
+
+interface ErrorType {
+  type: CustomErrors;
+  message: string;
+  metadata?: {
+    fields: Record<string, unknown>;
```

Comment:
I think this should just be `metadata?: Record<string, unknown>`, I understand what dom was saying now, I had no need for the specific field keys, at the time I thought I needed that but I could've worked without for my form errors on the client